packages/@sanity/schema/test/validation/validation.test.ts

```diff
@@ -89,4 +94,116 @@ describe('Validation test', () => {
       helpId: 'schema-standalone-block-type',
     })
   })
+
+  test('validate block members as object like', () => {
+    const schemaDef = [
+      {
+        title: 'Valid object',
+        name: 'validObject',
+        type: 'object',
+        fields: [
+          {
+            title: 'Blocks',
+            name: 'blocks',
+            type: 'array',
+            of: [{type: 'block', of: [{type: 'image', name: 'myImage'}]}],
+          },
+        ],
+      },
+      {
+        title: 'Invalid object',
+        name: 'invalidObject',
+        type: 'object',
+        fields: [
+          {
+            title: 'Blocks',
+            name: 'blocks',
+            type: 'array',
+            of: [
+              {
+                type: 'block',
+                of: [
+                  // Should produce error
+                  {type: 'string', name: 'foo'},
+                  // Should produce warning
+                  {type: 'object', name: 'validObject', fields: [{type: 'string', name: 'foo'}]},
+                  // Should be allowed
+                  {type: 'image', name: 'image'},
+                  // Should be allowed
+                  {type: 'reference', name: 'reference', to: {type: 'author'}},
+                  // Should produce warning
+                  {type: 'object', name: 'reference', fields: [{type: 'string', name: 'foo'}]},
+                  // Should produce warning
+                  {type: 'object', name: 'image', fields: [{type: 'string', name: 'foo'}]},
+                  // Should produce warning
+                  {type: 'object', name: 'file', fields: [{type: 'string', name: 'foo'}]},
+                  // Should produce warning
+                  {type: 'object', name: 'span', fields: [{type: 'string', name: 'foo'}]},
+                  // Should not be allowed
+                  {type: 'span', name: 'something', fields: [{type: 'string', name: 'foo'}]},
+                  // Should be allowed
+                  {type: 'reference', name: 'reference', to: {type: 'author'}},
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    ]
+    const coreTypes = [
```

Comment:
Could this be imported from [/src/sanity/coreTypes.ts](https://github.com/sanity-io/sanity/pull/4870/files#diff-985c6b5eb4dfacfe03546f5ebcd798bebb2bd29a562995de498c9a691b76b7e0) ?

Reply:
Yeah, this file was ready for a refactor that's for sure. Thanks for noticing :green_heart: 

I have added a suggestion in a new [commit](https://github.com/sanity-io/sanity/pull/4870/commits/0151177f3376af5549726f5e57c92d5cead42d59) that for sure removes a lot of this cruft, and just use the public `validateSchema` function instead.

I can't see any downsides (but there are upsides) by doing this instead?

@bjoerge - can you have another look?

Reply:
Thanks for this update and refactor. I can't think of any downside on using the public function, it's better to do it that way IMHO, if you add any breaking change to that public function the tests will catch it.